include/linker/devicetree_regions.h

```diff
@@ -43,21 +45,16 @@
  * @param node_id devicetree node identifier
  * @param attr region attributes
  */
-#define _REGION_DECLARE(node_id, attr)		    \
-	LINKER_DT_NODE_REGION_NAME(node_id)(attr) : \
-	ORIGIN = DT_REG_ADDR(node_id),		    \
+#define _REGION_DECLARE(node_id)	      \
+	LINKER_DT_NODE_REGION_NAME(node_id) : \
+	ORIGIN = DT_REG_ADDR(node_id),	      \
 	LENGTH = DT_REG_SIZE(node_id)
 
 /** @endcond */
 
 /**
- * @brief Generate a linker memory region from a devicetree node
- *
- * @param node_id devicetree node identifier with a \<reg\> property defining
- *                region location and size
- * @param attr region attributes to use (rx, rw, ...)
+ * @brief Generate linker memory regions from the device tree nodes with
+ *        compatible 'zephyr,memory-region'
  */
-#define LINKER_DT_REGION_FROM_NODE(node_id, attr)      \
-	COND_CODE_1(DT_NODE_HAS_STATUS(node_id, okay), \
-		    (_REGION_DECLARE(node_id, attr)),  \
-		    ())
+#define LINKER_DT_REGION() \
+	DT_FOREACH_STATUS_OKAY(_DT_COMPATIBLE, _REGION_DECLARE)
```

Comment:
```suggestion
	DT_FOREACH_STATUS_OKAY(_DT_COMPATIBLE, _REGION_DECLARE(attr))
```
I am not sure here, if it is correct suggestion, how to use attributes in this context.